lib/std/os/uefi/protocols/loaded_image_protocol.zig

```diff
@@ -0,0 +1,36 @@
+const uefi = @import("std").os.uefi;
+const Guid = uefi.Guid;
+const Handle = uefi.Handle;
+const SystemTable = uefi.tables.SystemTable;
+const MemoryType = uefi.tables.MemoryType;
+const DevicePathProtocol = uefi.protocols.DevicePathProtocol;
+
+pub const LoadedImageProtocol = extern struct {
+    revision: u32,
+    parent_handle: Handle,
+    system_table: *SystemTable,
+    device_handle: Handle,
```

Comment:
you sure this is supposed to be optional?

Reply:
`device_handle` is not explicitly specified as optional, but edk2 sets this to null if the image is loaded from memory.